Framework/DataHandling/src/LoadILLTOF.cpp

```diff
@@ -616,14 +609,16 @@ void LoadILLTOF::loadSpectra(size_t &spec, size_t firstSpec,
     for (size_t j = 0; j < m_numberOfPixelsPerTube; ++j) {
       if (spec > firstSpec) {
         // just copy the time binning axis to every spectra
-        m_localWorkspace->dataX(spec) = m_localWorkspace->readX(firstSpec);
+        m_localWorkspace->setSharedX(spec,
+                                     m_localWorkspace->sharedX(firstSpec));
```

Comment:
You removed an assignment above, shouldn't you also remove the `if` and change the source index?